Compile bug: (windows/vulkan/debug build) add_rms_data and add_rms_len not found

### Git commit

4da1e37513fdf749b01e2c1fe7b09cff43cf7e53

### Operating systems

Windows

### GGML backends

Vulkan

### Problem description & steps to reproduce

The release build compiles fine, the error only happens when building for debug.
I have been looking at the code and, from what I can make of it, the symbols mentioned should be required in release too. In a way, I'm more puzzled about why the release version does compile.

NB: I'm including llama.cpp in (via cmake, using FetchContent_xxx) in a larger project and this is why I sometimes compile it as debug.

### First Bad Commit

_No response_

### Compile command

```shell
#-------------------------
#  llama.cpp
#-------------------------
set( LLAMA_BUILD_COMMON ON )
#set( LLAMA_BUILD_TOOLS ON )
set( LLAMA_CURL OFF )
if(MSVC)
 set(GGML_BACKEND_DL ON)
 set(GGML_CUDA ON)
 set(GGML_VULKAN ON)
 set(GGML_CPU_ALL_VARIANTS ON)
endif()
FetchContent_Declare(
	llama
	GIT_REPOSITORY https://github.com/ggerganov/llama.cpp
	GIT_TAG        b6715
    EXCLUDE_FROM_ALL # don't automatically package all of llama.cpp in our distrib
)
```

### Relevant log output

```shell
3>C:\code\girolamo\build\_deps\llama-src\ggml\src\ggml-vulkan\ggml-vulkan.cpp(3445,5): error C2065: 'add_rms_len': undeclared identifier
3>C:\code\girolamo\build\_deps\llama-src\ggml\src\ggml-vulkan\ggml-vulkan.cpp(3445,5): error C2065: 'add_rms_data': undeclared identifier
3>C:\code\girolamo\build\_deps\llama-src\ggml\src\ggml-vulkan\ggml-vulkan.cpp(3445,5): error C2064: term does not evaluate to a function taking 9 arguments
3>    C:\code\girolamo\build\_deps\llama-src\ggml\src\ggml-vulkan\ggml-vulkan.cpp(3445,5):
3>    class does not define an 'operator()' or a user defined conversion operator to a pointer-to-function or reference-to-function that takes appropriate number of arguments
3>    C:\code\girolamo\build\_deps\llama-src\ggml\src\ggml-vulkan\ggml-vulkan.cpp(3445,5):
3>    while trying to match the argument list '(vk_device, vk_pipeline, std::basic_string<char,std::char_traits<char>,std::allocator<char>>, const char [5], int, size_t, initializer list, initializer list, int)'
3>C:\code\girolamo\build\_deps\llama-src\ggml\src\ggml-vulkan\ggml-vulkan.cpp(3446,5): error C2065: 'add_rms_len': undeclared identifier
3>C:\code\girolamo\build\_deps\llama-src\ggml\src\ggml-vulkan\ggml-vulkan.cpp(3446,5): error C2065: 'add_rms_data': undeclared identifier
3>C:\code\girolamo\build\_deps\llama-src\ggml\src\ggml-vulkan\ggml-vulkan.cpp(3446,5): error C2064: term does not evaluate to a function taking 9 arguments
3>    C:\code\girolamo\build\_deps\llama-src\ggml\src\ggml-vulkan\ggml-vulkan.cpp(3446,5):
3>    class does not define an 'operator()' or a user defined conversion operator to a pointer-to-function or reference-to-function that takes appropriate number of arguments
3>    C:\code\girolamo\build\_deps\llama-src\ggml\src\ggml-vulkan\ggml-vulkan.cpp(3446,5):
3>    while trying to match the argument list '(vk_device, vk_pipeline, std::basic_string<char,std::char_traits<char>,std::allocator<char>>, const char [5], int, size_t, initializer list, initializer list, int)'
```

--- a/ggml/src/ggml-vulkan/vulkan-shaders/vulkan-shaders-gen.cpp
+++ b/ggml/src/ggml-vulkan/vulkan-shaders/vulkan-shaders-gen.cpp
@@ -959,7 +959,7 @@ void write_output_files() {
     }
 
     std::string suffixes[2] = {"_f32", "_f16"};
-    for (auto op : {"add", "sub", "mul", "div", "add_rms"}) {
+    for (std::string op : {"add", "sub", "mul", "div", "add_rms"}) {
         hdr << "extern const void * " << op << "_data[2][2][2][2];\n";
         hdr << "extern const uint64_t " << op << "_len[2][2][2][2];\n";
 

